File: Code/PatternBattleGame/Assets/Scripts/GUIScripts/GamePlayGUIControl.js
#pragma strict

//Glyph Flyin Tracker
var GlyphIn : GameObject;
var MainGameGUI :GameObject;
var Symbols : GameObject;
var GlyphZoomSpeed = 1.0000;
@HideInInspector
var ScaleSmall = 1.00;
var ScaleLarge = 1.50;
@HideInInspector
var Zoom = false;
@HideInInspector
var Shrink = false;

@HideInInspector
var state = "";

var zoomSound : AudioClip;
var shrinkSound : AudioClip;

private var myAudioSource : AudioSource;


function Start () {
//	TurnGlyphsAnimationControllerOn(0f);
//	ZoomSmall();
//	Shrink = true;
	myAudioSource = gameObject.AddComponent.<AudioSource>();
	myAudioSource.volume = PlayerPrefs.GetInt("sound",1);
}


function FixedUpdate () {

ZoomLarge();
ZoomSmall();

}


function ZoomLarge(){

	if (!Zoom) {
		// so that we don't evaluate them all every frame.
		return;
	}

	if(Symbols.transform.localScale.x < ScaleLarge){
		Symbols.transform.localScale.x = Symbols.transform.localScale.x+(GlyphZoomSpeed*Time.deltaTime);
		Symbols.transform.localScale.y = Symbols.transform.localScale.y+(GlyphZoomSpeed*Time.deltaTime);
		
		if(Symbols.transform.localScale.x >= ScaleLarge){
			Zoom = false;
			doneZooming();
		}
	}
}


function ZoomSmall(){

	if (!Shrink) {
		// so that we don't evaluate them all every frame.
		return;
	}

	if(Symbols.transform.localScale.x > ScaleSmall){
		Symbols.transform.localScale.x = Symbols.transform.localScale.x-(GlyphZoomSpeed*Time.deltaTime);
		Symbols.transform.localScale.y = Symbols.transform.localScale.y-(GlyphZoomSpeed*Time.deltaTime);
		
		if(Symbols.transform.localScale.x <= ScaleSmall){
			Shrink = false;
			doneShrinking();
		}
	}
}

function doShrink(delay:float){
	Invoke("_doShrink",delay);
	//Debug.Log("doshrink with delay");
}

function doZoom(delay:float){
	Invoke("_doZoom",delay);
}

function _doZoom(){
	Zoom=true;
	Shrink=false;
	myAudioSource.PlayOneShot(zoomSound);
}

function _doShrink(){
	Shrink=true;
	Zoom=false;
	myAudioSource.PlayOneShot(shrinkSound);
	//Debug.Log("doshrink");
}


function doneZooming(){
	Debug.Log("done zooming");
}

function doneShrinking(){
	Debug.Log("done shrinking");
	gameObject.SendMessage("EndOfFlyIn");
}



function TurnGlyphsAnimationControllerOff(){
	if(this.GetComponent(ObjectRotate)!=0)
		this.GetComponent(ObjectRotate).last_turn = true;
	var rotateScripts = Symbols.GetComponentsInChildren(ObjectRotate);
	for (var r = 0 ; r < rotateScripts.Length ; r++) {
		var o :ObjectRotate = rotateScripts[r];
		if(o.RotationZ!=0)
			o.last_turn = true;
	}
}


function ForceStopRotating(){
	this.GetComponent(ObjectRotate).rotation_disable = true;
	var rotateScripts = Symbols.GetComponentsInChildren(ObjectRotate);
	for (var r = 0 ; r < rotateScripts.Length ; r++) {
		var o :ObjectRotate = rotateScripts[r];
		o.rotation_disable = true;
	}
}

function _TurnGlyphsAnimationControllerOn(){
	GlyphIn.GetComponent(ObjectRotate).start_rotating();
	var rotateScripts = Symbols.GetComponentsInChildren(ObjectRotate);
	for (var r = 0 ; r < rotateScripts.Length ; r++) {
		var o :ObjectRotate = rotateScripts[r];
		o.start_rotating();
	}
}

function TurnGlyphsAnimationControllerOn(delay:float){
	Invoke("_TurnGlyphsAnimationControllerOn",delay);
}

File: Code/PatternBattleGame/Assets/Scripts/GUIScripts/GamePlayGUIControl.cs
using JetBrains.Annotations;

using UnityEngine;

public class GamePlayGUIControl : MonoBehaviour
{
    public GameObject glyphIn;
    public GameObject mainGameGUI;
    public GameObject symbols;

    public float glyphZoomSpeed = 1f;

    [HideInInspector]
    public float scaleSmall = 1f;

    public float scaleLarge = 1.5f;

    [HideInInspector]
    public bool zoom;

    [HideInInspector]
    public bool shrink;

    [HideInInspector]
    public string state = "";

    public AudioClip zoomSound;
    public AudioClip shrinkSound;

    private AudioSource myAudioSource;

    [UsedImplicitly]
    private void Start()
    {
        myAudioSource = gameObject.AddComponent<AudioSource>();
        myAudioSource.volume = PlayerPrefs.GetInt("sound", 1);
    }

    [UsedImplicitly]
    private void Update()
    {
        ZoomLarge();
        ZoomSmall();
    }

    public void ZoomLarge()
    {
        if (!zoom)
        {
            return;
        }

        if (symbols.transform.localScale.x < scaleLarge)
        {
            Vector3 localScale = symbols.transform.localScale;
            localScale.x = symbols.transform.localScale.x + (glyphZoomSpeed * Time.deltaTime);
            localScale.y = symbols.transform.localScale.y + (glyphZoomSpeed * Time.deltaTime);

            symbols.transform.localScale = localScale;

            if (symbols.transform.localScale.x >= scaleLarge)
            {
                zoom = false;
                doneZooming();
            }
        }
    }

    public void ZoomSmall()
    {
        if (!shrink)
        {
            return;
        }

        if (symbols.transform.localScale.x > scaleSmall)
        {
            Vector3 localScale = symbols.transform.localScale;
            localScale.x = symbols.transform.localScale.x - (glyphZoomSpeed * Time.deltaTime);
            localScale.y = symbols.transform.localScale.y - (glyphZoomSpeed * Time.deltaTime);

            symbols.transform.localScale = localScale;

            if (symbols.transform.localScale.x >= scaleLarge)
            {
                shrink = false;
                doneShrinking();
            }
        }
    }

    public void doShrink(float delay)
    {
        Invoke("_doShrink", delay);
    }

    public void doZoom(float delay)
    {
        Invoke("_doZoom", delay);
    }

    public void _doZoom()
    {
        zoom = true;
        shrink = false;
        myAudioSource.PlayOneShot(zoomSound);
    }

    public void _doShrink()
    {
        shrink = true;
        zoom = false;
        myAudioSource.PlayOneShot(shrinkSound);
    }

    public void doneZooming()
    {
        Debug.Log("done zooming");
    }

    public void doneShrinking()
    {
        Debug.Log("done shrinking");
        gameObject.SendMessage("EndOfFlyIn");
    }

    public void TurnGlyphsAnimationControllerOff()
    {
        if (GetComponent<ObjectRotate>() != null)
            GetComponent<ObjectRotate>().lastTurn = true;
        var rotateScripts = symbols.GetComponentsInChildren<ObjectRotate>();
        for (var r = 0; r < rotateScripts.Length; r++)
        {
            ObjectRotate o = rotateScripts[r];
            if (o.rotationZ != 0)
                o.lastTurn = true;
        }
    }
    
    public void ForceStopRotating()
    {
        GetComponent<ObjectRotate>().rotationDisable = true;
        var rotateScripts = symbols.GetComponentsInChildren<ObjectRotate>();
        for (var r = 0; r < rotateScripts.Length; r++)
        {
            ObjectRotate o = rotateScripts[r];
            o.rotationDisable = true;
        }
    }

    public void _TurnGlyphsAnimationControllerOn()
    {
        glyphIn.GetComponent<ObjectRotate>().start_rotating();
        var rotateScripts = symbols.GetComponentsInChildren<ObjectRotate>();
        for (var r = 0; r < rotateScripts.Length; r++)
        {
            ObjectRotate o = rotateScripts[r];
            o.start_rotating();
        }
    }

    public void TurnGlyphsAnimationControllerOn(float delay)
    {
        Invoke("_TurnGlyphsAnimationControllerOn", delay);
    }
}